test/functional/interface_rpc.py

```diff
@@ -16,7 +16,8 @@ def expect_http_status(expected_http_status, expected_rpc_code,
                        fcn, *args):
     try:
         fcn(*args)
-        raise AssertionError(f"Expected RPC error {expected_rpc_code}, got none")
+        if expected_rpc_code is not None:
+            raise AssertionError(f"Expected RPC error {expected_rpc_code}, got none")
```

Comment:
In d11fb70b3fea7c1be5f48e32837d2c81dc7cbd10 "test: cover more HTTP error codes in interface_rpc.py"

This seems unnecessary?

Reply:
Reverting this change fails the test because we are divorcing HTTP status from RPC error, so you can have an HTTP OK with RPC error and some test cases now also ensure both are OK.

```
2023-07-07T15:44:27.147000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/Users/matthewzipkin/Desktop/work/bitcoin/test/functional/test_framework/test_framework.py", line 131, in main
    self.run_test()
  File "/Users/matthewzipkin/Desktop/work/bitcoin/test/functional/interface_rpc.py", line 157, in run_test
    self.test_http_status_codes()
  File "/Users/matthewzipkin/Desktop/work/bitcoin/test/functional/interface_rpc.py", line 125, in test_http_status_codes
    expect_http_status(200, None,   self.nodes[0].getblockhash, 0)
  File "/Users/matthewzipkin/Desktop/work/bitcoin/test/functional/interface_rpc.py", line 19, in expect_http_status
    raise AssertionError(f"Expected RPC error {expected_rpc_code}, got none")
AssertionError: Expected RPC error None, got none
```

Reply:
Hmm, the tests still seemed to work when I had run it..